src/taskcreator.js

```diff
@@ -79,7 +92,34 @@ class TaskCreator {
       this.lastCreateTask = {taskId: options.taskId, task};
       return {status: {taskId: options.taskId}};
     }
-    return await queue.createTask(options.taskId, task);
+
+    let lastFire, taskCreateRes;
+    try {
+      taskCreateRes = await queue.createTask(options.taskId, task);
+      lastFire = {
+        result: 'success',
+        taskId: hook.nextTaskId,
+        time: new Date(),
+      };
+    } catch (err) {
+      lastFire = {
+        result: 'error',
+        error: err,
+        time: new Date(),
+      };
+    }
+
+    try {
+      await appendLastFire(hook, context, lastFire);
+    } catch (err) {
+      debug('Failed to append lastfire with err: %s', err);
+      this.monitor.reportError(err);
+    }
+
+    if (lastFire.error) {
+      return Promise.reject(lastFire.error);
+    }
```

Comment:
I don't think this is necessary -- the error has already been recorded in the LastFire table *and* reported via the monitor, so we can just return success here.

Reply:
If there is an error notification may be send via email. https://github.com/taskcluster/taskcluster-hooks/blob/master/src/scheduler.js#L125

Reply:
Ah, good point.